### ISSUE TITLE
Express Backend

### ISSUE THREAD

**Issue by ovaltechnologiesinc** [neutralinojs/neutralinojs#899]
Is it possible to hook up an express server to a NEU app?
This is my stack: 
Vue JS for the front end and Express for the backend. I also am using SQLite for my database.

I am confused as to how to piece all these together to make a desktop app with success.
Any pointers, sample code and help will be appreciated.

**Comment by danidre14** [comment]
In short, the answer is *no*.

Express uses routes and feeds pages to the browser. Neu by nature just feeds the file in the `url` and `documentRoot` of the config file, which are local paths on the device, rather than routes like express.

Also, it will not be localhost safe even if we used the routes by express.

A feature request can be to change the `url` or `documentRoot` of the neu application without relaunching the application.

**Comment by Blatman** [comment]
It is possible to start an Express server as a packaged executable using the execCommand() API. eg: I use Caxa to bundle node with an Express server and the bundle ends up being c. 30M. The server can be set for localhost and whatever port and webroot. I haven't quite conquered the sqlite side of things but have been trying the execCommand method and also the Extension method (Discussion #737 and Issues #795). I think using the Extensions is possibly the best avenue for a database but I haven't yet managed to get an sqlite scenario working.

**Comment by danidre14** [comment]
@Blatman yeah it is possible to use Express that way. However, that Express will act as a server, and it cannot be integrated with Neutralino since they work in different ways:
Express works as a server that gets requests and feeds responses (html pages or otherwise)
Neutralino launches with only one index html file, with a mini server that performs the necessary operations on the computer.

You are right, Extensions can be the possible route.

I'm no SQLite professional, but if you can get SQLite to work in a normal NodeJS file (or python file), then it'll be a matter of hooking up the extensions to work in that file, using Neu's extensions API.

**Comment by pkulijing** [comment]
I've been doing something similar. I start my server as an extension of the neutralino app. The neutralino static server serves static files while my own server handles normal CRUD requests. Of course it would be better if the two web servers could be integrated into one, but the extension approach is enough for the moment. Just make sure you handle the CORS issue corretly in your own server because requests from the front end will be treated as cross-origin requests by your own server.

**Comment by ovaltechnologiesinc** [comment]
Thanks guys for your assistance. Please can you hack up a sample code to
demonstrate this process for me? I will appreciate it so much. Thanks once
again.
